workers/unity/Assets/Fps/Scripts/GameLogic/WorldTiles/TileTypeCollection.cs

```diff
@@ -0,0 +1,23 @@
+﻿using UnityEngine;
+
+namespace Fps
+{
+    [CreateAssetMenu(fileName = "New Prefab Collection", menuName = "Improbable/Create Prefab Collection")]
+    public class TileTypeCollection : ScriptableObject
+    {
+        [SerializeField] private GameObject[] Tiles;
+        [Range(0f, 1f)] public float ChanceOfEmptyTile;
+        public Color DisplayColor = Color.blue;
+
+        public GameObject GetRandomTile()
+        {
+            var rand = Random.value;
+            if (ChanceOfEmptyTile == 0f || rand > ChanceOfEmptyTile)
```

Comment:
Is the first part of this statement strictly necessary?

Reply:
It is; if `rand == 0f`, and `ChanceOfEmptyTile == 0f`, then `rand > ChanceOfEmptyTile` is false and so it makes an empty tile despite ChanceOfEmptyTile being zero. There might be a better way to do this, though.

Reply:
Why not `rand >= ChanceOfEmtpyTile` then? 

Also `==` operations on floating points are always a little funky

Reply:
aha, because if `rand == 1.0f` and `ChanceOfEmptyTile == 1.0f`, then it will not make an empty tile - and it should!